nepalingo-web/src/components/user_auth/User_auth.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from "react";
+import supabase from "./supabaseClient";
+import { useNavigate } from "react-router-dom";
+
+const User_auth: React.FC = () => {
```

Comment:
btw have you seen this react component? https://supabase.com/docs/guides/auth/auth-helpers/auth-ui